storm-core/src/jvm/backtype/storm/messaging/netty/SaslStormServerHandler.java

```diff
@@ -136,17 +135,19 @@ public void messageReceived(ChannelHandlerContext ctx, MessageEvent e)
 
     @Override
     public void exceptionCaught(ChannelHandlerContext ctx, ExceptionEvent e) {
-        server.closeChannel(e.getChannel());
+        if(server != null) server.closeChannel(e.getChannel());
     }
 
     private void getSASLCredentials() throws IOException {
-        topologyName = (String) this.server.storm_conf
-                .get(Config.TOPOLOGY_NAME);
-        String secretKey = SaslUtils.getSecretKey(this.server.storm_conf);
+        String secretKey;
+        topologyName = server.name();
+        secretKey = server.secretKey();
+
         if (secretKey != null) {
             token = secretKey.getBytes();
         }
+
         LOG.debug("SASL credentials for storm topology " + topologyName
-                + " is " + secretKey);
+                  + " is " + secretKey);
```

Comment:
Let's use format strings in this log message so we don't construct strings every time.